### PR TITLE
Remove PVM Instance Id from deployment file

### LINKED ISSUE #1: Remove hardcoded value of PVM instance id(Service instance ID) for node and controller plugin deployment files
The value for service instance id is hard coded in the deployment files of node and controller plugins.
Need to remove the hardcoded values and fetch the value automatically from the script.

### PR DESCRIPTION
**Pull request by Madhan-SWE**
Service instance ID mentioned as PVM Instance Id in the deployment file is removed by below steps:
   

- Each of node of the Kubernetes is added with labels for service instance id, node instance id and region
- Since Service instance id is collected from the run time from node label, PVM instance id is not needed in the deployment file.

fixes #1

### CODE REVIEW

**Comment by Madhan-SWE** [comment]
@mkumatag  can you review this?

I am not able to link the issue here as the link option is disabled for me.

File: pkg/cloud/metadata.go [hunk 1]
```diff
@@ -0,0 +1,101 @@
...
+		if val, ok := labels[key]; ok {
+			switch key {
+			case "serviceInstanceId":
+				instanceInfo.serviceInstanceID = val
+				break
+			case "nodeInstanceId":
+				instanceInfo.nodeInstanceId = val
+				break
+			case "region":
+				instanceInfo.region = val
```
Comment (mkumatag) [inline pkg/cloud/metadata.go:92]: see if we can use some of the standard labels from here - https://kubernetes.io/docs/reference/labels-annotations-taints/#topologykubernetesiozone

File: pkg/cloud/metadata.go [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+	keysList := []string{"serviceInstanceId", "nodeInstanceId", "region"}
+	instanceInfo := Metadata{}
+	for _, key := range keysList {
+		if val, ok := labels[key]; ok {
+			switch key {
+			case "serviceInstanceId":
+				instanceInfo.serviceInstanceID = val
+				break
+			case "nodeInstanceId":
+				instanceInfo.nodeInstanceId = val
```
Comment (mkumatag) [inline pkg/cloud/metadata.go:89]: look into other providers how are they setting the provider-specific labels.

File: pkg/cloud/metadata.go [hunk 3]
```diff
@@ -0,0 +1,101 @@
...
+				break
+			case "region":
+				instanceInfo.region = val
+				break
```
Comment (mkumatag) [inline pkg/cloud/metadata.go:93]: in general all the switch cases should be declared as const variables and then be used here.

File: pkg/cloud/metadata.go [hunk 4]
```diff
@@ -0,0 +1,101 @@
+package cloud
```
[the remark this answers is not in the document]
Comment (mkumatag) [reply pkg/cloud/metadata.go:1]: having that said, we can keep this as is.

File: deploy/kubernetes/base/serviceaccount-csi-node.yaml [hunk 5]
```diff
@@ -0,0 +1,8 @@
+---
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: powervs-csi-node-sa
+  namespace: kube-system
+  labels:
+    app.kubernetes.io/name: powervs-csi-driver
```
Comment (mkumatag) [inline deploy/kubernetes/base/serviceaccount-csi-node.yaml:8]: add a newline in the end

File: go.mod [hunk 6]
```diff
@@ -10,7 +10,10 @@ require (
 	github.com/dgrijalva/jwt-go v3.2.0+incompatible
 	github.com/golang/glog v0.0.0-20160126235308-23def4e6c14b
 	google.golang.org/grpc v1.27.0
+	k8s.io/api v0.19.4
 	k8s.io/apimachinery v0.19.4
+	k8s.io/client-go v0.19.4
+	k8s.io/klog v1.0.0
```
Comment (mkumatag) [inline go.mod:16]: where is this v1 used, lets use the `k8s.io/klog/v2` instead

File: pkg/cloud/powervs.go [hunk 7]
```diff
@@ -185,7 +181,7 @@ func (p *powerVSCloud) GetPVMInstanceByName(name string) (*PVMInstance, error) {
 		return nil, err
 	}
 	for _, pvmInstance := range in.PvmInstances {
-		if name == *pvmInstance.ServerName {
+		if name == strings.ToLower(*pvmInstance.ServerName) {
```
Comment (mkumatag) [inline pkg/cloud/powervs.go:184]: is there a specific reason why we are converting this lower?

**Review by mkumatag — CHANGES_REQUESTED** [review]
initial review comments

File: pkg/cloud/metadata.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (Madhan-SWE) [reply pkg/cloud/metadata.go:1]: This service doesn't connect to power cloud, this is gonna be running inside the nodes or pods and then collect node labels using Kubernetes APIs.

File: pkg/cloud/powervs.go (hunk 7 above)
Comment (mkumatag) [reply pkg/cloud/powervs.go:184]: will ibm cloud respect case in that case? what happens when user creates VM with `sample-vm` and `Sample-vm`

File: go.mod (hunk 6 above)
Comment (Madhan-SWE) [reply go.mod:16]: fixed

File: pkg/cloud/metadata.go (hunk 2 above)
Comment (Madhan-SWE) [reply pkg/cloud/metadata.go:89]: this is the format I found from EBS CSI Driver,
```
topology.ebs.csi.aws.com/zone
```

Comment (Madhan-SWE) [reply pkg/cloud/metadata.go:89]: Few other formats from VMware CSI Driver:
```
        // Appplatform label that all vDPP PVCs must have.
	appplatformLabel = "appplatform.vmware.com/instance-id"
	// Opt out label present on pod to look for bound PVCs of all sibling replicas
	siblingReplicaCheckOptOutLabel = "psp.vmware.com/sibling-replica-check-opt-out"
	// Selected node annotation present on PVC
	pvcSelectedNode = "volume.kubernetes.io/selected-node"
	
```

Comment (mkumatag) [reply pkg/cloud/metadata.go:89]: xref: https://github.com/kubernetes/kubernetes/blob/ae9ca48f01ddb03731e7903cfe91ef3db9ce8990/staging/src/k8s.io/api/core/v1/well_known_labels.go#L27

Comment (Madhan-SWE) [reply pkg/cloud/metadata.go:89]: I suggest we can use something like,

```
cloud.ibm.com/cloud-instance-id
cloud.ibm.con/pvm-instance-id
```

File: pkg/cloud/metadata.go (hunk 3 above)
Comment (Madhan-SWE) [reply pkg/cloud/metadata.go:93]: Fixed

File: pkg/cloud/powervs.go (hunk 7 above)
Comment (mkumatag) [reply pkg/cloud/powervs.go:184]: probably we can create a request with powervs team to have these constraints integrated as well.

**Comment by Madhan-SWE** [comment]
@mkumatag  Review changes added. Please approve and merge the PR.

File: block/pod.yaml [hunk 8]
```diff
@@ -14,4 +14,5 @@ spec:
   volumes:
     - name: data
       persistentVolumeClaim:
-        claimName: block-claim
\ No newline at end of file
+        claimName: block-claim
+
```
Comment (mkumatag) [inline block/pod.yaml:18]: not exactly a empty new line, you can refer https://stackoverflow.com/questions/5813311/whats-the-significance-of-the-no-newline-at-end-of-file-log to understand what I'm talking about

**Review by mkumatag — CHANGES_REQUESTED** [review]
Small nit, otherwise lgtm

File: block/pod.yaml (hunk 8 above)
Comment (Madhan-SWE) [reply block/pod.yaml]: fixed

**Review by mkumatag — APPROVED** [review]
lgtm

**Comment by mkumatag** [comment]
this project doesn't have OWNERS file yet, hence needs a manual merge.

**Pull request merged**